pkg/cache/cluster.go

```diff
@@ -652,7 +657,7 @@ func (c *clusterCache) IterateHierarchy(key kube.ResourceKey, action func(resour
 					return strings.Compare(key1.String(), key2.String()) < 0
 				})
 				child := children[0]
-				action(child, nsNodes)
+				action(nil, child, nsNodes)
 				child.iterateChildren(nsNodes, map[kube.ResourceKey]bool{res.ResourceKey(): true}, action)
```

Comment:
I think it is better to keep `circular dependency` error internal to gitops-engine. Can you remove `error` parameter `cluster.IterateHierarchy` method and keep it in `resource.iterateChildren` only. E.g. : 

```
child.iterateChildren(nsNodes, map[kube.ResourceKey]bool{res.ResourceKey(): true}, func(err error, child *Resource, namespaceResources map[kube.ResourceKey]*Resource) {
	if err != nil {
		c.log.V(2).Info(err.Error()) 
	}
	action(child, namespaceResources)
})
```